packages/ui-kit/src/components/AccessTokenProvider/AccessTokenProvider.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { createContext, PropsWithChildren, useCallback, useEffect, useRef, useState } from 'react'
+import { useLog } from '../Log'
+
+export interface AccessTokenContextValue {
+  /**
+   * The returned accessToken wheter from the props or from the fetchToken function
+   */
+  accessToken?: string
+  /**
+   * If true, the access token is being fetched
+   */
+  isLoading?: boolean
+  /**
+   * Function that will be called when the access token expires.
+   */
+  onExpiredToken?: () => void
+  /**
+   * If true, the access token failed to be fetched after the maximum retries
+   */
+  failedRetry?: boolean
+}
+
+const ACCESS_TOKEN_REFRESH_INTERVAL = 3300000 // 55 minutes
+const ACCESS_TOKEN_MAX_RETRIES = 3
+
+export const AccessTokenContext = createContext<AccessTokenContextValue | undefined>({})
+
+export interface AccessTokenProviderProps extends PropsWithChildren {
+  /**
+  * Function that the provider will use to fetch the access token.
+  * @returns {Promise<string>} A promise that resolves to the access token
+  */
+  fetchToken?: () => Promise<string>
+  /**
+  * If passed, the provider will ignore the `fetchToken` function and pass this access token to all the children components.
+  */
+  accessToken?: string
+  /**
+  * Function that will be called when the access token expires.
+  */
+  onAccessTokenExpired?: () => void
+}
+
+export const AccessTokenProvider = (props: AccessTokenProviderProps) => {
+  const { children, accessToken: accessTokenFromProps, fetchToken, onAccessTokenExpired } = props
```

Comment:
Not need to rename accessToken as accessTokenFromProps. We can use it directly as accessToken in AccessTokenProvider method.

Let's simplify it as "accessToken"

Reply:
I think here I would keep it as `accessTokenFromProps` since it could get confusing between the `fetchedToken` and the `accessTokenFromProps`, I just want it to be clear which one is the fetched one and which one comes from props, then the `accessToken` is the one that will be served, does that make sense?

@markandrus wdyt about this?

Reply:
I think this one is probably more opinionated, I'm going to change it for now if in a future we think other names could fit better we can change it 👍